src/ServantElm.hs

```diff
@@ -166,6 +229,42 @@ mkRequest request =
 
     expect =
       case request ^. reqReturnType of
-        Just _ ->
-          "Http.expectWhatever toMsg"
+        Just elmTypeExpr ->
+          "Http.expectJson toMsg" <+> (typeRefDecoder . definitionToRef) elmTypeExpr
         Nothing -> error "mkHttpRequest: no reqReturnType?"
+
+    body =
+      case request ^. reqBody of
+        Just _ ->
+          "Http.jsonBody bodyValue"
```

Comment:
Use the to-be extracted `bodyValue` constant here.

Reply:
done